mods/tnt/init.lua

```diff
@@ -1,11 +1,6 @@
 tnt = {}
--- Default to enabled in singleplayer and disabled in multiplayer
-local singleplayer = minetest.is_singleplayer()
-local setting = minetest.setting_getbool("enable_tnt")
-if (not singleplayer and setting ~= true) or
-		(singleplayer and setting == false) then
-	return
-end
+-- Default to enabled
+local enable_tnt = minetest.setting_getbool("enable_tnt") ~= false
```

Comment:
Background to that:

The previous if statement was logically redundant and had no effect at all without the return (but even with the return it could simply have been an `if setting` without changing behavior. Singleplayer or not did not make a difference before.)

For example the second part of the statement would set `enable_tnt`  only to **`false`** if it already were **`false`**.
The first part when it was either **`false`**  as well, or when **`nil`** in multiplayer (which would make no difference to the enabled state later on, since **`false`** and **`nil`** are treated here the same later)

The `~=false` is new though, which defaults everything to enabled, if there is no setting available. If you wanted to change its fallback depending on singleplayer/multiplayer, you could easier do:

```lua
local enable_tnt = minetest.setting_getbool("enable_tnt") or minetest.is_singleplayer()
```

Personally i don't mind to what it is defaulting, but the if-statement itself was just confusing and not doing anything anymore.